src/Router.js

```diff
@@ -110,12 +110,19 @@ export default class Router {
   }
 
   /**
-   * @param {function} fn - Function to execute while popstate listener is disabled
+   * @param {function} fn - Async function to execute w/o popstate listener
    */
   executeWithoutPopstateListener(fn) {
     this.__shouldHandlePopstateEvents = false;
-    fn();
-    this.__shouldHandlePopstateEvents = true;
+
+    // Execute the function, then re-enable the popstate listener
+    return fn().then((result) => {
```

Comment:
I imagine you are using the two function arguments with `.then` for best cross compatibility with jQuery Deferreds and Promises?  This is probably best since this is a module used by potentially disparate types of FE clients.

Reply:
True that it would make this more compatible with deferreds, but considering the following reasoning, we can probably go with `.finally` as James has suggested:

- In the `.then(resolveHander, rejectHandler)` setup, I was still returning `Promise.<resolve|reject>()`, so it would still add some weirdness if using deferreds with this
- Promises are preferred as we are trying to move away from jQuery deferreds

Reply:
As we discussed offline, I'm reverting back to the original solution (`.then(resolveHander, rejectHandler)`) since the lack of `.finally` polyfill causes CI tests to fail.